crates/client/src/worker_registry/mod.rs

```diff
@@ -78,15 +79,18 @@ impl ClientWorkerSetImpl {
         task_queue: String,
     ) -> Option<SlotReservation> {
         let key = SlotKey::new(namespace, task_queue);
-        if let Some(p) = self.slot_providers.get(&key)
-            && let Some(worker) = self.all_workers.get(p)
-            && let Some(slot) = worker.try_reserve_wft_slot()
-        {
-            let deployment_options = worker.deployment_options();
-            return Some(SlotReservation {
-                slot,
-                deployment_options,
-            });
+        if let Some(worker_list) = self.slot_providers.get(&key) {
+            let worker_id = worker_list.choose(&mut rand::rng());
+            if let Some(worker_entry) = worker_id
+                && let Some(worker) = self.all_workers.get(&worker_entry.0)
+                && let Some(slot) = worker.try_reserve_wft_slot()
```

Comment:
In principle we should then try the next worker in the list if this reservation fails. In practice I doubt this is going to come up much. Probably not hard to add, but maybe annoying to test.